forest/src/cli/sync_cmd.rs

```diff
@@ -0,0 +1,188 @@
+// Copyright 2020 ChainSafe Systems
+// SPDX-License-Identifier: Apache-2.0, MIT
+
+use super::stringify_rpc_err;
+use actor::EPOCH_DURATION_SECONDS;
+use chain_sync::get_naive_time_now;
+use chrono::naive::NaiveDateTime;
+use chrono::prelude::*;
+use cid::Cid;
+use jsonrpc_v2::Error as JsonRpcError;
+use jsonrpsee::raw::RawClient;
+use jsonrpsee::transport::http::HttpTransportClient as HTC;
+use rpc_client::{check_bad, head, mark_bad, new_client, status, submit_block};
+use std::thread;
+use std::time::Duration;
+use structopt::StructOpt;
+
+#[derive(Debug, StructOpt)]
+pub enum SyncCommand {
+    #[structopt(
+        name = "mark-bad",
+        about = "Mark the given block as bad, will prevent syncing to a chain that contains it"
+    )]
+    MarkBad {
+        #[structopt(help = "Block Cid given as string argument")]
+        block_cid: String,
+    },
+
+    #[structopt(
+        name = "check-bad",
+        about = "Check if the given block was marked bad, and for what reason"
+    )]
+    CheckBad {
+        #[structopt(help = "Block Cid given as string argument")]
+        block_cid: String,
+    },
+
+    #[structopt(
+        name = "submit",
+        about = "Submit newly created block to network through node"
+    )]
+    Submit {
+        #[structopt(help = "Gossip block as String argument")]
+        gossip_block: String,
+    },
+
+    #[structopt(name = "status", about = "Check sync status")]
+    Status,
+
+    #[structopt(name = "wait", about = "Wait for sync to be complete")]
+    Wait,
+}
+
+fn get_naive_time_zero() -> NaiveDateTime {
+    NaiveDate::from_ymd(1, 1, 1).and_hms(0, 0, 0)
+}
+
+impl SyncCommand {
+    pub async fn run(self) {
+        let mut client = new_client();
+
+        match self {
+            SyncCommand::Status {} => {
```

Comment:
This command also doesn't fail for any reason when there isn't a node running (nothing is output)